packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -378,8 +412,14 @@ class TextPainter {
   }
 
   /// Paints the text onto the given canvas at the given offset.
+  ///
+  /// Valid only after [layout] or [layoutToMaxIntrinsicWidth] has been called.
   void paint(Canvas canvas, Offset offset) {
-    assert(!_needsLayout && "Please call layout() before paint() to position the text before painting it." is String);
+    assert(() {
+      if (_needsLayout)
+        throw new FlutterError('Please call layout() before paint() to position the text before painting it.');
```

Comment:
Add `'TextPainter.paint called when text geometry was not yet calculated.\n'` before the 'Please...' text.